demo/src/checkout-pricing.js

```diff
@@ -15,24 +15,41 @@ function CheckoutPricingForm () {
   const [recurlyError, setRecurlyError] = useState(null);
   const [plan, setPlan] = useState('');
   const [quantity, setQuantity] = useState('');
+  const [coupon, setCoupon] = useState('');
+  const [giftCard, setGiftCard] = useState('');
+  const [currency, setCurrency] = useState('');
+  const [billingCountry, setBillingCountry] = useState('');
+  const [billingPostalCode, setBillingPostalCode] = useState('');
+  const [billingVatNumber, setBillingVatNumber] = useState('');
+  const [shippingCountry, setShippingCountry] = useState('');
+  const [shippingPostalCode, setShippingPostalCode] = useState('');
+  const [shippingVatNumber, setShippingVatNumber] = useState('');
   const [{ price, loading }, setPricing] = useCheckoutPricing({}, setRecurlyError);
+  const showPrice = price && price.now && !recurlyError
 
   function updatePlan (e) {
     const plan = e.target.value;
-    setRecurlyError(null);
     setPlan(plan);
-    setPricing(prev => ({ ...prev, subscriptions: [{ plan, quantity }] }));
   };
 
   function updateQuantity (e) {
     const quantity = e.target.value;
+    setQuantity(quantity);
+  };
+
+  function handleSubmit (e) {
+    e.preventDefault();
+    const address = {
+      billingCountry,
+      billingPostalCode,
+      billingVatNumber,
+    };
```

Comment:
collapsing the address state tracking as above would allow this to be direct assignment

```js
setPricing({ subscriptions: [{ plan, quantity }], coupon, giftCard, address: billingAddress });
```